### PR TITLE
Update ServletJSPAPI to the Jakarta family (covers up to Jakarta EE 8)

### PR DESCRIPTION
**Pull request by pepness**
- Change license type to EPL v2.0
- Use one license file for the external jars and one for the generated jar
- Use jakarta family jars
- Update servlet-api from 3.1 to 4.0
- Update jsp-api from 2.3.3 to 2.3.6
- Update gf-client-module from 5.0 to 5.1.0
- Remove unused versions of servlet-api, they weren't included in the
  generated.jar
- Change servlet version from 3.1 to 4.0 in the name of the generated
  jar and all references to it
- Remove xsd files that do not belong to the javax naming
  (up to Jakarta EE 8)
- Override missing methods from Servlet 4.0
- Add new entry to ignore-overlaps for the jakarta.servlet-api jar

NetBeans Testing:

- Verify successful execution of libraries and licenses Ant test
- Verify successful execution of Verify Sigtests
- Verify successful execution of unit tests for modules servletjspapi, web.core, web.core.syntax, web.el, web.jsf.editor, web.jspparser, web.project
- Verify that the jars are in the Ant Library
- Started NetBeans and ensure the log didn't have any ERROR or new WARNINGS
- Successfully test web apps with jsp and servelts and verify that it works.
jsp-compilation

### CODE REVIEW

File: enterprise/web.jspparser/src/org/netbeans/modules/web/jspparser/ParserServletContext.java [hunk 1]
```diff
@@ -660,6 +660,12 @@ public Dynamic addServlet(String string, Servlet srvlt) {
     public Dynamic addServlet(String string, Class<? extends Servlet> type) {
         return null;
     }
+    
+    @Override
+    public Dynamic addJspFile(String servletName, String jspFile) {
+        System.err.println("** addJspFile(servletName,jspFile)");
```
Comment (matthiasblaesing) [inline enterprise/web.jspparser/src/org/netbeans/modules/web/jspparser/ParserServletContext.java:666]: Should this be a logger or even completely removed?

File: enterprise/web.jspparser/src/org/netbeans/modules/web/jspparser/ParserServletContext.java [hunk 2]
```diff
@@ -293,7 +293,7 @@ public Enumeration getInitParameterNames() {
      */
     public int getMajorVersion() {
         
-        return (3);
+        return (4);
```
Comment (matthiasblaesing) [inline enterprise/web.jspparser/src/org/netbeans/modules/web/jspparser/ParserServletContext.java:296]: The parentheses are not necessary

**Review by matthiasblaesing — APPROVED** [review]
In general looks like a good cleanup and seems sensible to me.

The sig file shows removals of members, but as this is imported API and only a low probability to actually hit a problem.

I left two inline comments, which are not really problematic.

File: enterprise/web.jspparser/src/org/netbeans/modules/web/jspparser/ParserServletContext.java (hunk 2 above)
Comment (pepness) [reply enterprise/web.jspparser/src/org/netbeans/modules/web/jspparser/ParserServletContext.java:296]: Done.

File: enterprise/web.jspparser/src/org/netbeans/modules/web/jspparser/ParserServletContext.java (hunk 1 above)
Comment (pepness) [reply enterprise/web.jspparser/src/org/netbeans/modules/web/jspparser/ParserServletContext.java]: It is not necessary.

**Review by matthiasblaesing — APPROVED** [review]

**Pull request merged**